
syntax = 'proto3';

service Multiplayer {
  rpc UpdatePlayer (PlayerUpdate) returns (PlayerUpdateResponse) {}
}

message PlayerUpdate {
  PlayerProto player = 1;
  PlayerShootEvent shoot = 2;
}

message PlayerUpdateResponse {
  repeated PlayerProto player = 1;
}

message PlayerShootEvent {
  bool exists = 1;
  float dirX = 2;
  float dirY = 3;
  float dirZ = 4;
  float dirW = 5;
}

message PlayerProto {
  Transform transform = 1;
}

message Transform {
  Vector3 position = 1;
  Vector4 rotation = 2;

  message Vector3 {
    float x = 1;
    float y = 2;
    float z = 3;
  }
  message Vector4 {
    float x = 1;
    float y = 2;
    float z = 3;
    float w = 4;
  }
}
